embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/config/source/CombinedRemoteConfigSource.kt

```diff
@@ -5,22 +5,20 @@ import io.embrace.android.embracesdk.internal.config.store.RemoteConfigStore
 import io.embrace.android.embracesdk.internal.worker.BackgroundWorker
 import java.util.concurrent.TimeUnit
 
-internal class CombinedRemoteConfigSource(
+class CombinedRemoteConfigSource(
     private val store: RemoteConfigStore,
     private val httpSource: RemoteConfigSource,
     private val worker: BackgroundWorker,
     private val intervalMs: Long = 60 * 60 * 1000
 ) {
 
     // the remote config that is used for the lifetime of the process.
-    private val cfg = store.loadConfig()
+    private val response by lazy { store.loadResponse() }
 
-    fun getConfig(): RemoteConfig? = cfg
+    fun getConfig(): RemoteConfig? = response?.cfg
 
     fun scheduleConfigRequests() {
-        worker.submit {
-            store.retrieveEtag()?.let(httpSource::setInitialEtag)
-        }
+        response?.etag?.let(httpSource::setInitialEtag)
         worker.scheduleWithFixedDelay(
```

Comment:
Do we even want to schedule this with a repeated job? Why not just schedule the job for an hour later when it completes? Seems unnecessarily complicated to maintain that extra bit of state